pkg/storage/unified/resource/server.go

```diff
@@ -436,19 +429,37 @@ func (s *server) newEvent(ctx context.Context, user claims.AuthInfo, key *Resour
 		return nil, err
 	}
 
-	// We only set name for update checks
-	if check.Verb == utils.VerbUpdate {
-		check.Name = key.Name
-	}
+	// For folder moves, we need to check permissions on both folders
+	if s.isFolderMove(event) {
+		if event.ObjectOld == nil {
+			return nil, NewBadRequestError("missing old object")
+		}
```

Comment:
This check feels unneeded (`isFolderMove` already checks for this), but also happy to keep it in if you prefer.

Reply:
I agree that it should always be nil ... but If felt wrong not to double check in case isFolderMove gets updated